### PR TITLE
Generator and cnn

### PR DESCRIPTION
**Pull request by ModusMorris**
-  **Aligned step labels**: Shifted labels to ensure CNN predicts steps **at peak positions** instead of early/late detections.
- -  **Adaptive peak thresholding**: Implemented dynamic thresholding based on mean probability to detect both weak and strong steps.
- **Increased dropout**: Strengthened dropout to `0.3` to improve generalization and reduce false positives.
-  **Overall effect**: More precise step detections, fewer misaligned predictions, and better model robustness to different walking patterns.
Training with 5 Epochs:
image
image

### CODE REVIEW

**Comment by AlhaririAnas** [comment]
Great work! I'm currently working on some adjustments

**Comment by ModusMorris** [comment]
very good, you are welcome to continue working on the branch, if you have any questions please ask me

**Comment by AlhaririAnas** [comment]
### Changes Made:

- Added Evaluation Metrics:

  - Included **classification_report, confusion_matrix, and ROC curve** to evaluate model performance.
    - classification_report:
      grafik


    - confusion_matrix:
    
       Figure_2

    
    - ROC curve:
    
       Figure_3


  - Added `evaluate_model` function to compute and display these metrics.


- Training Improvements:

  - Added validation loss tracking during training to monitor overfitting.
     - 38 datasets, total: 206290 samples.
Train samples: 165032, Test samples: 41258
Epoch 1/5: 100%|█████████████████████████████████| 5158/5158 [00:25<00:00, 199.27it/s]
🔵 Epoch 1, Train Loss: 1.7823, Test Loss: 1.5440

  - Added loss curve plotting as well.
     Figure_1


- Code Refactoring:

  - Made the code more modular and added comments for clarity.

**Comment by AlhaririAnas** [comment]
@ModusMorris I have reviewed your code and everything looks good. Additionally I have made my adjustments, could you please review them and if you agree please tell me to merge the branch into main.

**Comment by ModusMorris** [comment]
okay looks pretty good, just not quite sure what we need this for and one question i see you removed the cnn once and added it back

**Comment by ModusMorris** [comment]
that you have made the code modular and clearer is great

**Comment by ModusMorris** [comment]
the result looks a bit strange at the end. The ground truth has completely shifted here, you'll have to take a look to make sure it's like it was before 
image

**Comment by AlhaririAnas** [comment]
> okay looks pretty good, just not quite sure what we need this for and one question i see you removed the cnn once and added it back

We use the classification report and ROC curve to evaluate the model's performance, especially by looking at the F1 score. This helps us compare different models or adjustments to see if they improve performance.

The loss rate is included to check for overfitting or underfitting.

Regarding the files, I changed the filename, and after pushing, I noticed that there were now two files - one with the new name and one with the old name. To avoid duplication, I deleted the old file.

**Comment by AlhaririAnas** [comment]
> the result looks a bit strange at the end. The ground truth has completely shifted here, you'll have to take a look to make sure it's like it was before
> image

I wonder why the ground truth values have shifted. Everything looks fine on my end. We can talk about it later in meeting.

grafik

**Comment by ModusMorris** [comment]
my mistake had still loaded a wrong csv
image
now its right, everything looks good we can merge it :)
